keps/sig-cli/2953-kustomize-plugin-graduation/README.md

```diff
@@ -745,6 +765,17 @@ It makes it easier for Kustomize users to invoke third party code, which can be
 * Graduate "KRM Function style" plugins as described (including removing starlark and further restricting exec), but do not introduce Catalog or Composition.
   * This would be a less usable and useful plugin system than the one proposed, but better than nothing. Catalog and Composition are encapsulated enough that plugin GA can proceed without them if they fail or lose contributor bandwidth before completion. Changes to the flag strategy would be required.
 
+
+_Alternative for the handling of plugins that are not included in a Catalog:_
+
+Instead of disallowing uncatalogued plugins, we could introduce a series of flags to govern them specifically. This is not recommended because it complicates the overall mental model of how plugins work, and because the flags themselves would likely be unwieldy to use. That said, this is how such an alternative could work.
+
+By default, plugins not listed in a trusted catalog will not be executed and an error will be printed. The providers for these plugins must be listed explicitly in each plugin config, and additional flags are always required to use them. Which flags are required depends on the provider type.
+
+For plugins distributed as containers, the `--enable-container-plugins` flag will need to be provided. The containers will continue to be denied disk and network access by default. To enable access, the existing network/mount restriction flags will be retained but renamed to clarify what they govern:  `--plugin-container-networks=[STRING]` `--plugin-container-mounts=[STRING]`.
```

Comment:
Is the network flag a binary flag? If so, maybe using `--enable-plugin-container-networks` is better?

Reply:
The current flag is a named network: ` --network-name string`